src/code/z_player_lib.c

```diff
@@ -1824,46 +1824,156 @@ Gfx gCullFrontDList[] = {
+    gLinkHumanEyesLeftTex,    // PLAYER_EYES_LEFT
+    gLinkHumanEyesUpTex,      // PLAYER_EYES_UP
+    gLinkHumanEyesDownTex,    // PLAYER_EYES_DOWN
+    gLinkHumanEyesWincingTex, // PLAYER_EYES_WINCING
+};
+
+static TexturePtr sMouthTextures[PLAYER_MOUTH_MAX] = {
     gLinkHumanMouthClosedTex, // PLAYER_MOUTH_CLOSED
-    gLinkHumanMouthTeethTex,  // PLAYER_MOUTH_TEETH
-    gLinkHumanMouthAngryTex,  // PLAYER_MOUTH_ANGRY
-    gLinkHumanMouthHappyTex,  // PLAYER_MOUTH_HAPPY
-};
-
-typedef struct PlayerFaceIndices {
-    /* 0x0 */ u8 eyeIndex;
-    /* 0x1 */ u8 mouthIndex;
-} PlayerFaceIndices; // size = 0x2
-
-PlayerFaceIndices sPlayerFaces[] = {
-    { PLAYER_EYES_OPEN, PLAYER_MOUTH_CLOSED },       // PLAYER_FACE_0
-    { PLAYER_EYES_HALF, PLAYER_MOUTH_CLOSED },       // PLAYER_FACE_1
-    { PLAYER_EYES_CLOSED, PLAYER_MOUTH_CLOSED },     // PLAYER_FACE_2
-    { PLAYER_EYES_OPEN, PLAYER_MOUTH_CLOSED },       // PLAYER_FACE_3
-    { PLAYER_EYES_HALF, PLAYER_MOUTH_CLOSED },       // PLAYER_FACE_4
-    { PLAYER_EYES_CLOSED, PLAYER_MOUTH_CLOSED },     // PLAYER_FACE_5
-    { PLAYER_EYES_ROLL_LEFT, PLAYER_MOUTH_CLOSED },  // PLAYER_FACE_6
-    { PLAYER_EYES_ROLL_UP, PLAYER_MOUTH_TEETH },     // PLAYER_FACE_7
-    { PLAYER_EYES_7, PLAYER_MOUTH_ANGRY },           // PLAYER_FACE_8
-    { PLAYER_EYES_OPEN, PLAYER_MOUTH_ANGRY },        // PLAYER_FACE_9
-    { PLAYER_EYES_ROLL_RIGHT, PLAYER_MOUTH_CLOSED }, // PLAYER_FACE_10
-    { PLAYER_EYES_ROLL_LEFT, PLAYER_MOUTH_CLOSED },  // PLAYER_FACE_11
-    { PLAYER_EYES_CLOSED, PLAYER_MOUTH_ANGRY },      // PLAYER_FACE_12
-    { PLAYER_EYES_HALF, PLAYER_MOUTH_TEETH },        // PLAYER_FACE_13
-    { PLAYER_EYES_OPEN, PLAYER_MOUTH_ANGRY },        // PLAYER_FACE_14
-    { PLAYER_EYES_OPEN, PLAYER_MOUTH_HAPPY },        // PLAYER_FACE_15
+    gLinkHumanMouthHalfTex,   // PLAYER_MOUTH_HALF
+    gLinkHumanMouthOpenTex,   // PLAYER_MOUTH_OPEN
+    gLinkHumanMouthSmileTex,  // PLAYER_MOUTH_SMILE
+};
+#else
+static TexturePtr sEyeTextures[PLAYER_FORM_MAX][PLAYER_EYES_MAX] = {
+    // PLAYER_FORM_FIERCE_DEITY
+    {
+        NULL, // PLAYER_EYES_OPEN
+        NULL, // PLAYER_EYES_HALF
+        NULL, // PLAYER_EYES_CLOSED
+        NULL, // PLAYER_EYES_RIGHT
+        NULL, // PLAYER_EYES_LEFT
+        NULL, // PLAYER_EYES_UP
+        NULL, // PLAYER_EYES_DOWN
+        NULL, // PLAYER_EYES_WINCING
+    },
+    // PLAYER_FORM_GORON
+    // Note: use PLAYER_EYES_WINCING to access `gLinkGoronEyesSurprisedTex`. See `Player_DrawImpl`.
+    {
+        gLinkGoronEyesOpenTex,      // PLAYER_EYES_OPEN
+        gLinkGoronEyesHalfTex,      // PLAYER_EYES_HALF
+        gLinkGoronEyesClosedTex,    // PLAYER_EYES_CLOSED
+        gLinkGoronEyesSurprisedTex, // PLAYER_EYES_RIGHT
+        NULL,                       // PLAYER_EYES_LEFT
+        NULL,                       // PLAYER_EYES_UP
+        NULL,                       // PLAYER_EYES_DOWN
+        NULL,                       // PLAYER_EYES_WINCING
```

Comment:
I don't think goron should have eye textures. Half are NULLs and some others don't even match the offset in the object, which makes this an actual change instead of a shiftability fix

Reply:
If you look at the xmls, then these do actually match the offsets:
```c
<Texture Name="gLinkGoronEyesOpenTex" Offset="0x0"
<Texture Name="gLinkGoronEyesHalfTex" Offset="0x800" 
<Texture Name="gLinkGoronEyesClosedTex" Offset="0x1000"
<Texture Name="gLinkGoronEyesSurprisedTex" Offset="0x1800"
```
and
```c
<Texture Name="gLinkHumanEyesOpenTex" Offset="0x0"
<Texture Name="gLinkHumanEyesHalfTex" Offset="0x800"
<Texture Name="gLinkHumanEyesClosedTex" Offset="0x1000"
<Texture Name="gLinkHumanEyesRightTex" Offset="0x1800"
```

And to manage the NULL entries, there's special code in `Player_DrawImpl` i.e.

```c
if (playerForm == PLAYER_FORM_GORON) {
    // Goron does not have the eye textures to look in different directions
    if ((eyeIndex >= PLAYER_EYES_RIGHT) && (eyeIndex <= PLAYER_EYES_DOWN)) {
        eyeIndex = PLAYER_EYES_OPEN;
    } else if (eyeIndex == PLAYER_EYES_WINCING) {
        // Goron form puts a surpised expression where the eyes-right normally goes
        eyeIndex = PLAYER_EYES_RIGHT;
    }
}
```

which maps all 8 eyeIndex values to valid eyeIndex values:
```c
PLAYER_EYES_OPEN    -> PLAYER_EYES_OPEN   (gLinkGoronEyesOpenTex)
PLAYER_EYES_HALF    -> PLAYER_EYES_HALF   (gLinkGoronEyesHalfTex)
PLAYER_EYES_CLOSED  -> PLAYER_EYES_CLOSED (gLinkGoronEyesClosedTex)
PLAYER_EYES_RIGHT   -> PLAYER_EYES_OPEN   (gLinkGoronEyesOpenTex)
PLAYER_EYES_LEFT    -> PLAYER_EYES_OPEN   (gLinkGoronEyesOpenTex)
PLAYER_EYES_UP      -> PLAYER_EYES_OPEN   (gLinkGoronEyesOpenTex)
PLAYER_EYES_DOWN    -> PLAYER_EYES_OPEN   (gLinkGoronEyesOpenTex)
PLAYER_EYES_WINCING -> PLAYER_EYES_RIGHT  (gLinkGoronEyesSurprisedTex)
```

Reply:
Oh I see. It is so silly to do half of it normally and to special case the other half.